src/clue.js

```diff
@@ -2,27 +2,162 @@
+  color: 'green',
+  },{
+  // drOrchid = {
+  firstName: 'Doctor',
+  lastName: 'Orchid',
+  occupation: 'Scientist',
+  age: 26,
+  description: 'PhD in plant toxicology. Adopted daughter of Mr. Boddy',
+  image: 'http://www.radiotimes.com/uploads/images/Original/111967.jpg',
+  color: 'white',
+  },{
+  // profPlum = {
+  firstName: 'Victor',
+  lastName: 'Plum',
+  occupation: 'Designer',
+  age: 22,
+  description: 'Billionaire video game designer',
+  image: 'https://66.media.tumblr.com/ee7155882178f73b3781603f0908617c/tumblr_phhxc7EhPJ1w5fh03_540.jpg',
+  color: 'purple',
+  },{
+  // missScarlet = {
+  firstName: 'Kasandra',
+  lastName: 'Scarlet',
+  occupation: 'Actor',
+  age: 31,
+  description: 'She is an A-list movie star with a dark past',
+  image: 'https://www.radiotimes.com/uploads/images/Original/111967.jpg',
+  color: 'red',
+  },{
+  // mrsPeacock = {
+  firstName: 'Eleanor',
+  lastName: 'Peacock',
+  occupation: 'Socialité',
+  age: 36,
+  description: 'She is from a wealthy family and uses her status and money to earn popularity',
+  image: 'https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg',
+  color: 'blue',
+  },{
+  // mrMustard = {
+  firstName: 'Jack',
+  lastName: 'Mustard',
+  occupation: 'Retired Football player',
+  age: 62,
+  description: 'He is a former football player who tries to get by on his former glory',
+  image: 'https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg',
+  color: 'yellow',
+}];
 
 // Rooms Array
 
-const roomsArray = [];
+const roomsArray = [
+
+  { name: 'Dining Room', },
+  { name: 'Conservatory', },
+  { name: 'Kitchen', },
+  { name: 'Study', },
+  { name: 'Library', },
+  { name: 'Billiard Room', },
+  { name: 'Lounge', },
+  { name: 'Ballroom', },
+  { name: 'Hall', },
+  { name: 'Spa', },
+  { name: 'Living Room', },
+  { name: 'Observatory', },
+  { name: 'Theater', },
+  { name: 'Guest House', },
+  { name: 'Patio', },
+];
 
 // Weapons Array
 
-const weaponsArray = [];
+const weaponsArray = [
+  {
+    name: 'rope',
+    weight: 10,
+  }, {
+    name: 'knife',
+    weight: 8,
+  }, {
+    name: 'candlestick',
+    weight: 2,
+  }, {
+    name: 'dumbbell',
+    weight: 30,
+  }, {
+    name: 'poison',
+    weight: 2,
+  }, {
+    name: 'axe',
+    weight: 15,
+  }, {
+    name: 'bat',
+    weight: 13,
+  }, {
+    name: 'trophy',
+    weight: 25,
+  }, {
+    name: 'pistol',
+    weight: 20,
+  }];
 
 
 // ITERATION 2
 
-function selectRandom() {}
+function selectRandom(array) {
+  if (array.length === 0) {
+    return undefined;
+  }
+  let i = Math.floor(Math.random() * array.length);
+  return array[i];
+}
+
+
+function pickMystery() {
+
+  let suspect = suspectsArray[Math.floor(Math.random() * suspectsArray.length)];
+
+  let room = roomsArray[Math.floor(Math.random() * roomsArray.length)];
+
+  let weapon = weaponsArray[Math.floor(Math.random() * weaponsArray.length)];
+
+  return { suspect, weapon, room };
+
```

Comment:
hier kannst du die selectRandom Funktion wiederverwenden! Ansonsten ergibt die ja sozusagen gar keinen wirklichen Sinn. Beispielsweise so:
```
function pickMystery() {

  let suspect = selectRandom(suspectsArray)

  let room = selectRandom(roomsArray)

  let weapon = selectRandom(weaponsArray)

  return { suspect, weapon, room };

}
```

Man kann das auch noch alles vorher in einem Objekt abspeichern und dann nur das Objekt zurückgeben, aber so funktioniert es natürlich auch.